raster/src/main/scala/geotrellis/raster/geotiff/GeoTiffReprojectRasterSource.scala

```diff
@@ -112,22 +112,31 @@ class GeoTiffReprojectRasterSource(
       queryPixelBounds <- bounds
       targetPixelBounds <- queryPixelBounds.intersection(this.dimensions)
     } yield {
+      val targetExtent = gridExtent.extentFor(targetPixelBounds)
       val targetRasterExtent = RasterExtent(
-        extent = gridExtent.extentFor(targetPixelBounds, clamp = true),
+        extent = targetExtent,
+        cols = targetPixelBounds.width.toInt,
+        rows = targetPixelBounds.height.toInt
+      )
+
+      // buffer the targetExtent to read a buffered area from the source tiff
+      // so the resample would behave properly on borders
+      val bufferedTargetRasterExtent = RasterExtent(
+        extent = targetExtent.buffer(cellSize.width, cellSize.height),
         cols = targetPixelBounds.width.toInt,
         rows = targetPixelBounds.height.toInt
       )
 
       // A tmp workaround for https://github.com/locationtech/proj4j/pull/29
       // Stacktrace details: https://github.com/geotrellis/geotrellis-contrib/pull/206#pullrequestreview-260115791
-      val sourceExtent = Proj4Transform.synchronized(targetRasterExtent.extent.reprojectAsPolygon(backTransform, 0.001).getEnvelopeInternal)
-      val sourcePixelBounds = closestTiffOverview.rasterExtent.gridBoundsFor(sourceExtent, clamp = true)
+      val sourceExtent = Proj4Transform.synchronized(bufferedTargetRasterExtent.extent.reprojectAsPolygon(backTransform, 0.001).getEnvelopeInternal)
+      val sourcePixelBounds = closestTiffOverview.rasterExtent.gridBoundsFor(sourceExtent)
       (sourcePixelBounds, targetRasterExtent)
     }}.toMap
 
     geoTiffTile.crop(intersectingWindows.keys.toSeq, bands.toArray).map { case (sourcePixelBounds, tile) =>
       val targetRasterExtent = intersectingWindows(sourcePixelBounds)
-      val sourceRaster = Raster(tile, closestTiffOverview.rasterExtent.extentFor(sourcePixelBounds, clamp = true))
+      val sourceRaster = Raster(tile, closestTiffOverview.rasterExtent.extentFor(sourcePixelBounds))
```

Comment:
Was removing clamp=true here critical to the fix? I'm having trouble thinking through the implications of doing this so curious to verify.

Reply:
oh, the default value of clamp is true, right.

Reply:
@echeipesh it is not, it is just confusing for me to read clamp=true and this is a default behavior. I can roll it back.